pkg/net/client/client.go

```diff
@@ -436,7 +435,7 @@ awaitfinished:
 	// Confirm that we actually ended up with the root block we wanted, failure
 	// here indicates a data transfer error that was not properly reported
 	if _, err := linkSystem.StorageReadOpener(ipld.LinkContext{}, cidlink.Link{Cid: rootCid}); err != nil {
-		if !errors.Is(err, format.ErrNotFound{}) { // may or may not get a go-ipld-format/ErrNotFound for not-found
+		if nf, ok := err.(interface{ NotFound() bool }); !ok || !nf.NotFound() {
```

Comment:
oh the joy

Reply:
we'll get there .. I'm still working on this upstream